### PR TITLE
Feature: S2S auth service

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Design API](https://tools.hmcts.net/jira/browse/RPE-435)

### Change description ###

Including basic S2S authorisation service used commonly across all reform platform services

Drags along basic exception handler for REST response

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/feature/ResponseExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @ExceptionHandler(InvalidTokenException.class)
+    protected ResponseEntity<Void> handleInvalidTokenException(InvalidTokenException exc) {
+        log.warn(exc.getMessage(), exc);
+        return status(UNAUTHORIZED).build();
+    }
+
+    @ExceptionHandler(JsonProcessingException.class)
+    protected ResponseEntity<String> handleJsonProcessingException() {
+        return status(BAD_REQUEST).body("Exception occurred while parsing letter contents");
+    }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/feature/ResponseExceptionHandler.java]: looks like some copy-pasted, unrelated code. remove

**Comment by rkondratowicz** [comment]
is there a design / documentation how this service is going to work?

**Comment by rkondratowicz** [comment]
will the clients be able to switch the toggle via API?
if not, is there a need to prevent service A from reading values of service's B toggles?

**Comment by doncem** [comment]
Switching is not allowed for clients - all amending is left for administrators which at the moment is under spike

Features are client specific (e.g. shutter page). Considering the way ff4j provides the database storage, we will have to prefix service specific features with it's name. Solves both problems: service A cannot access non-global feature of service B and we have a non-colliding feature support

Design/Documentation is in progress as project is under development/clarification of how and what. We've been discussing in several sessions with @nitinprabhu perhaps after the spike he's on we will be in better position to put that in writing

**Comment by doncem** [comment]
This has big probability of not being a part of project requirement anymore as apparently it is possible to assign role access which was not clear at the beginning. Will close later
